### PR TITLE
feat: add tenant specific feature flags

### PR DESCRIPTION
**Pull request by fenos**
## What kind of change does this PR introduce?

Feature

## What is the current behavior?

none

## What is the new behavior?

- Add any feature flag to specific tenant
- added `imageTransformation` feature flag
- in a multitenant environment, only tenants with the  `imageTransformation`  flag to true can transform images

### CODE REVIEW

File: src/database/tenant.ts [hunk 1]
```diff
@@ -114,6 +134,15 @@ export async function getFileSizeLimit(tenantId: string): Promise<number> {
   return fileSizeLimit
 }
 
+/**
+ * Get the file size limit from the tenant config
```
Comment (darora) [inline src/database/tenant.ts]: inaccurate

File: src/http/plugins/tenant-feature.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import fastifyPlugin from 'fastify-plugin'
+import { getConfig } from '../../config'
+import { Features, getFeatures } from '../../database/tenant'
+
+/**
+ * Middleware that requires a specific feature to be enabled
+ * for a given tenant.
+ *
+ * This only applies for multi-tenant-application.
+ * for single tenant, use environment variables to toggle features
```
Comment (darora) [inline src/http/plugins/tenant-feature.ts]: nits: 
- `For` (capitalized)
- consistent use of full stops to end sentences
- consistent hyphenation of multi-tenant and single-tenant

File: src/storage/backend/s3.ts [hunk 3]
```diff
@@ -22,7 +22,7 @@ import { StorageBackendError } from '../errors'
 
 /**
  * S3Backend
- * Interacts with a s3 system with this S3Backend adapter
+ * Interacts with an s3 compatible file system with this S3Adapter
```
Comment (darora) [inline src/storage/backend/s3.ts]: nit: hyphenate `s3-compatible`

File: src/storage/limits.ts [hunk 4]
```diff
@@ -16,6 +16,21 @@ export async function getFileSizeLimit(tenantId: string): Promise<number> {
   return fileSizeLimit
 }
 
+/**
+ * Determines if the image transformation feature should be enabled.
+ * Support multi tenancy
```
Comment (darora) [inline src/storage/limits.ts]: 'Support multi-tenancy' is a little confusing; almost sounds like a todo. 

nits: 
- consistent hyphenation for multi-tenancy

File: src/http/routes/tenant/index.ts [hunk 5]
```diff
@@ -112,14 +152,18 @@ export default async function routes(fastify: FastifyInstance) {
           file_size_limit: fileSizeLimit,
           jwt_secret: jwtSecret !== undefined ? encrypt(jwtSecret) : undefined,
           service_key: serviceKey !== undefined ? encrypt(serviceKey) : undefined,
+          feature_image_transformation:
```
Comment (darora) [inline src/http/routes/tenant/index.ts]: is this essentially `features?.imageTransformation?.enabled`?

**Review by darora — APPROVED** [review]

**Review by inian — APPROVED** [review]

**Pull request merged**